Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -125,4 +125,24 @@ Vector<String> transcodeImages(const Vector<String>& paths, const String& destin
     });
 }
 
+std::pair<String, Vector<IntSize>> utiAndAvailableSizesFromImageData(std::span<const uint8_t> data)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
+    auto uti = imageDecoder->uti();
+    if (!isSupportedImageType(imageDecoder->uti()))
+        return { { }, { } };
+
+    // No support for animated image.
+    if (imageDecoder->repetitionCount() == RepetitionCountInfinite)
+        return { uti, { } };
+
+    Vector<IntSize> sizes;
```

Comment:
We know the capacity of this vector will be the frame count, so we should reserve the initial capacity for better efficiency.